### PR TITLE
if ceph-disk fails to activate an OSD then bubble up the error

### PR DESCRIPTION
**Pull request by alfredodeza**
Fixes issue #670

### CODE REVIEW

File: roles/ceph-osd/tasks/activate_osds.yml [hunk 1]
```diff
@@ -20,11 +20,18 @@
     - devices
   changed_when: false
   failed_when: false
+  register: activate_osd_disk
   when:
     not item.0.get("skipped") and
     item.0.get("rc", 0) != 0 and
     not osd_auto_discovery
 
+- name: Fail if ceph-disk cannot create an OSD
```
Comment (leseb) [inline roles/ceph-osd/tasks/activate_osds.yml]: just to stay consistent can you do: `s/Fail/fail/`?
Thanks!

The rest looks good :)

**Comment by alfredodeza** [comment]
test this please

File: roles/ceph-osd/tasks/activate_osds.yml [hunk 2]
```diff
@@ -20,11 +20,18 @@
...
+  register: activate_osd_disk
   when:
     not item.0.get("skipped") and
     item.0.get("rc", 0) != 0 and
     not osd_auto_discovery
 
+- name: fail if ceph-disk cannot create an OSD
+  fail:
+    msg: "ceph-disk failed to create an OSD"
+  when: "ceph-disk: Error: ceph osd create failed" in item.get("stderr", "")
```
Comment (andrewschoen) [inline roles/ceph-osd/tasks/activate_osds.yml]: why not ``activate_osd_disk.get('stderr', '')``?  I don't think you need ``with_items`` here do you?

If you want all the lines of stderr I think you can use ``activate_osd_disk.stderr_lines``.

Comment (alfredodeza) [inline roles/ceph-osd/tasks/activate_osds.yml]: you can't because `activate_osd_disk` is actually a list because it gets captured from a loop.

**Pull request merged**